internal/alloydbutil/engine.go

```diff
@@ -85,16 +86,20 @@ func (p *PostgresEngine) Close() {
 // getUser retrieves the username, a flag indicating if IAM authentication
 // will be used and an error.
 func getUser(ctx context.Context, config engineConfig) (string, bool, error) {
-	// If neither user nor password are provided, retrieve IAM email.
-	if config.user == "" && config.password == "" {
+	if config.user != "" && config.password != "" {
+		// If both username and password are provided use provided username.
+		return config.user, false, nil
+	} else if config.iamAccountEmail != "" {
```

Comment:
I'm unclear on why this is getting updated -- what value is iamAccountEmail providing?

Reply:
Followed up offline: This is matching the python engine: https://github.com/googleapis/langchain-google-alloydb-pg-python/blob/99f48bc395511dfba50f5a222fb3a648041a0235/src/langchain_google_alloydb_pg/engine.py#L167